src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/FieldFixupSignature.cs

```diff
@@ -56,7 +56,7 @@ public override void AppendMangledName(NameMangler nameMangler, Utf8StringBuilde
         public override int CompareToImpl(ISortableNode other, CompilerComparer comparer)
         {
             FieldFixupSignature otherNode = (FieldFixupSignature)other;
-            int result = _fixupKind.CompareTo(otherNode._fixupKind);
+            int result = _fixupKind.CompareEnum(otherNode._fixupKind);
```

Comment:
```suggestion
            int result = ((int)_fixupKind).CompareTo((int)otherNode._fixupKind);
```

Reply:
Why not just `_fixupKind - otherNode._fixupKind` ?

Reply:
No need to call `CompareTo`. Just do the subtraction here directly as @huoyaoyuan suggests (and typecast if necessary)

Reply:
Note that the subtraction does not work to compare arbitrary integers. It only works if the integers are small. It is probably the case here, but then you need to think about all the time about whether you are dealing with small integers or not.

Reply:
Might be an [opportunity](https://gcc.godbolt.org/z/SpP7Lq) for changes in the JIT here: `(a == b) ? 0 : -Convert.ToInt32(a > b)` should be branch-free (`xor eax, eax; cmp edi, esi; setg al; neg eax`), or `Convert.ToInt32(a > b) - Convert.ToInt32(a < b)` should be something like `xor eax, eax; xor ecx,ecx; cmp edi, esi; seg al; setl cl; sub eax, ecx`. [ This doesn't seem to happen on a cursory glance.](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunGBLAHYYGAWVIAKQcOxoGUhsACUHLjzXEA7A2wMAvLoUMA/AwAMDENoYA+QyfIWGccgG5V3AL7vecoaPKSfjK+wkoq1GpqfABmDOI6tmGaDK7eUbHxDAA8CsrJzm4RkdzJpoVqXkUl/H4idIHSsvJhnFWRySwQAgBuMFAYAHQAKhAAkkJ0Egm5Tqxdvf3DYxNT2bnlPJXqNcIiKA3aTX4taTzJOnAKG57e9CGiAKwHwc3KrcUlWjr6dmaOcJ0en1BiNxhhJpkckprgxKpUgA==)

Reply:
Might be an opportunity to improve the codegen for `int.CompareTo()`.  It's currently not branch-free.  For instance, these two variants that works with arbitrary integers could be used instead of the branchy version:

- `(a == b) ? 0 : -Convert.ToInt32(a > b)` should be branch-free (`xor eax, eax; cmp edi, esi; setg al; neg eax`)
- `Convert.ToInt32(a > b) - Convert.ToInt32(a < b)` should be something like `xor eax, eax; xor ecx,ecx; cmp edi, esi; seg al; setl cl; sub eax, ecx`

(Might also be able to use a conditional move, although it doesn't seem that necessary.)